### PR TITLE
AF-593: Decouple kie-wb-common-datamodel from Drools

### PR DESCRIPTION
**Pull request by mbarkley**

### CODE REVIEW

**Comment by mbarkley** [comment]
Related:
https://github.com/AppFormer/uberfire/pull/822
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/541
https://github.com/kiegroup/droolsjbpm-knowledge/pull/263
https://github.com/kiegroup/drools/pull/1444
https://github.com/kiegroup/jbpm/pull/973
https://github.com/kiegroup/droolsjbpm-integration/pull/1163
https://github.com/kiegroup/kie-wb-common/pull/1133
https://github.com/kiegroup/jbpm-designer/pull/654
https://github.com/kiegroup/jbpm-wb/pull/864
https://github.com/kiegroup/optaplanner-wb/pull/218

File: drools-wb-extensions/drools-wb-drl-extension/drools-wb-drl-extension-server/src/main/java/org/kie/workbench/common/services/datamodel/backend/server/cache/DSLExtension.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline drools-wb-extensions/drools-wb-drl-extension/drools-wb-drl-extension-server/src/main/java/org/kie/workbench/common/services/datamodel/backend/server/cache/DSLExtension.java]: I believe the ```(C)``` is surplus to requirements and not part of the _standard_ licence.

File: drools-wb-extensions/drools-wb-drl-extension/drools-wb-drl-extension-server/src/test/java/org/kie/workbench/common/services/datamodel/backend/server/PackageDataModelDSLTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
+/*
+* Copyright 2010 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline drools-wb-extensions/drools-wb-drl-extension/drools-wb-drl-extension-server/src/test/java/org/kie/workbench/common/services/datamodel/backend/server/PackageDataModelDSLTest.java]: 2017

File: drools-wb-extensions/drools-wb-drl-extension/drools-wb-drl-extension-server/src/test/java/org/kie/workbench/common/services/datamodel/backend/server/PackageDataModelDSLTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import org.drools.workbench.models.datamodel.oracle.DSLActionSentence;
+import org.drools.workbench.models.datamodel.oracle.DSLConditionSentence;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.kie.workbench.common.services.datamodel.backend.server.builder.packages.PackageDataModelOracleBuilder;
+
+import static junit.framework.Assert.*;
+
+/**
+ * TODO
```
Comment (manstis) [inline drools-wb-extensions/drools-wb-drl-extension/drools-wb-drl-extension-server/src/test/java/org/kie/workbench/common/services/datamodel/backend/server/PackageDataModelDSLTest.java]: I'd prefer for this to be implemented in this PR.. I'm guessing this _was_ **to do** but can _now_ **be done**?

Comment (mbarkley) [reply drools-wb-extensions/drools-wb-drl-extension/drools-wb-drl-extension-server/src/test/java/org/kie/workbench/common/services/datamodel/backend/server/PackageDataModelDSLTest.java]: Oops. That was an oversight, and was absolutely meant to be done for this PR.

**Comment by mbarkley** [comment]
@manstis Fixed the license headers and rewrote the tests for the DSLExtension.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Pull request merged**